scripts/viz/param_sweep_latent_space_viz.R

```diff
@@ -0,0 +1,321 @@
+
+tybalt_best_model_png <- file.path(tyb_fig, "z_parameter_tybalt_best.png")
+tybalt_best_model_pdf <- file.path(tyb_fig, "z_parameter_tybalt_best.pdf")
+
+p <- ggplot(tybalt_good_training_df, aes(x = train_epoch, y = loss)) +
+  geom_line(aes(color = num_components, linetype = loss_type), size = 0.2) +
+  xlab("Training Epoch") +
+  ylab("Loss") +
+  scale_color_manual(name = "Latent Dimensions",
+                     values = c("#e41a1c", "#377eb8", "#4daf4a", "#984ea3",
+                                "#ff7f00", "#ffff33")) +
+  scale_linetype_manual(name = "Loss Type", values = c("solid", "dotted"),
+                        labels = c("Train", "Validation")) +
+  theme_bw() +
+  base_theme +
+  theme(axis.text.x = element_text(angle = 0))
+
+ggsave(tybalt_best_model_png, plot = p, height = 2.5, width = 4)
+ggsave(tybalt_best_model_pdf, plot = p, height = 2.5, width = 4)
+
+# ADAGE visualizations
+# Load ADAGE results, process and recode variables for plotting
+adage_df <- readr::read_tsv(adage_file,
+                            col_types = readr::cols(
+                              .default = readr::col_character(),
+                              train_epoch = readr::col_integer(),
+                              loss = readr::col_double(),
+                              val_loss = readr::col_double()))
+
+adage_melt_df <- reshape2::melt(adage_df,
+                                id.vars = c("learning_rate", "batch_size",
+                                            "epochs", "noise", "train_epoch",
+                                            "num_components", "sparsity",
+                                            "seed"),
+                                measure.vars = c("loss", "val_loss"),
+                                variable.name = "loss_type",
+                                value.name = "loss")
+
+adage_select_df <- adage_melt_df %>%
+  dplyr::filter(loss_type == "val_loss") %>%
+  dplyr::group_by(learning_rate, batch_size, epochs, noise, sparsity,
+                  num_components) %>%
+  dplyr::summarize(end_loss = tail(loss, 1))
+
+adage_select_df$noise <-
+  dplyr::recode_factor(adage_select_df$noise, 
+                       `0.0` = "Noise: 0",
+                       `0.1` = "Noise: 0.1",
+                       `0.5` = "Noise: 0.5")
+
+adage_select_df$learning_rate <-
+  dplyr::recode_factor(adage_select_df$learning_rate, 
+                       `0.0005` = "Learning: 0.0005",
+                       `0.001` = "Learning: 0.001", 
+                       `0.0015` = "Learning: 0.0015",
+                       `0.002` = "Learning: 0.002",
+                       `0.0025` = "Learning: 0.0025")
+
+# 1) Describe final validation loss for all hyperparameter combinations
+# across number of latent space dimensionality (x axis) for ADAGE
+adage_param_z_png <- file.path(adage_fig, "z_parameter_adage.png")
+adage_param_z_pdf <- file.path(adage_fig, "z_parameter_adage.pdf")
+
+p <- ggplot(adage_select_df,
```

Comment:
Can you reorder the levels in `sparsity` for this plot so that the legend is in increasing order? It took me a moment to square the plot with the text around it in `z_dimensions_hyperparameter_sweep_results.md` with the current ordering

Reply:
Yeah, good point. I will update the ordering in the next commit!